test_files/super.ts

```diff
@@ -0,0 +1,25 @@
+class SuperTestBase {
+  constructor(public x: number) {}
+}
+
+// A ctor with a parameter property.
+class SuperTestDerivedParamProps extends SuperTestBase {
+  constructor(public y: string) {
+    super(3);
+  }
+}
+
+// A ctor with an initialized property.
+class SuperTestDerivedInitializedProps extends SuperTestBase {
+  y: string = 'foo';
+  constructor() {
+    super(3);
+  }
+}
+
+// A ctor with a super() but none of the above two details.
+class SuperTestDerivedOrdinary extends SuperTestBase {
```

Comment:
Yes, it was mostly useful for comparing the output of all three before I simplified the logic to not special case the other two.  I could remove it I guess?  But it is one of the only instances of `super` in this code...